compiler/rustc_middle/src/values.rs

```diff
@@ -31,12 +32,12 @@ impl<'tcx> Value<TyCtxt<'tcx>> for ty::SymbolName<'_> {
     }
 }
 
-impl<'tcx> Value<TyCtxt<'tcx>> for ty::Binder<'_, ty::FnSig<'_>> {
-    fn from_cycle_error(tcx: TyCtxt<'tcx>, stack: &[QueryInfo]) -> Self {
+impl<'tcx> Value<TyCtxt<'tcx>, crate::dep_graph::DepKind> for ty::Binder<'_, ty::FnSig<'_>> {
```

Comment:
Can use `use crate::dep_graph::DepKind` here in the file? There are a lot of mentions of it.

Reply:
Implemented ­— In addition to this, many `D: DepKind`s have been replaced with the concrete DepKind type.